File: src/HMS.Domain/Entities/Appointment.cs
using HMS.Domain.Common;
using HMS.Domain.Enums;

namespace HMS.Domain.Entities;

public class Appointment : BaseEntity
{
    public Guid PatientId { get; set; }
    public Patient Patient { get; set; } = null!;

    public Guid DoctorId { get; set; }
    public Doctor Doctor { get; set; } = null!;

    public DateTime AppointmentDate { get; set; }
    public TimeSpan AppointmentTime { get; set; }
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;
    public string? ReasonForVisit { get; set; }
    public string? Notes { get; set; }
}

File: frontend/src/types/appointment.ts
export enum AppointmentStatus {
  Pending = 1,
  Confirmed = 2,
  InProgress = 3,
  Completed = 4,
  Cancelled = 5,
  NoShow = 6,
}

export const APPOINTMENT_STATUS_LABELS: Record<AppointmentStatus, string> = {
  [AppointmentStatus.Pending]: "Pending",
  [AppointmentStatus.Confirmed]: "Confirmed",
  [AppointmentStatus.InProgress]: "In Progress",
  [AppointmentStatus.Completed]: "Completed",
  [AppointmentStatus.Cancelled]: "Cancelled",
  [AppointmentStatus.NoShow]: "No Show",
};

export interface AppointmentDto {
  id: string;
  patientId: string;
  patientName: string;
  doctorId: string;
  doctorName: string;
  specialization: string;
  appointmentDate: string;
  appointmentTime: string;
  status: AppointmentStatus;
  reasonForVisit: string | null;
  notes: string | null;
  createdAt: string;
}

export interface CreateAppointmentDto {
  patientId: string;
  doctorId: string;
  appointmentDate: string;
  appointmentTime: string;
  reasonForVisit?: string;
}

export interface UpdateAppointmentDto {
  appointmentDate: string;
  appointmentTime: string;
  status: AppointmentStatus;
  reasonForVisit?: string;
  notes?: string;
}
